### PR TITLE
Fixed OMP for matrix.rms()

### LINKED ISSUE #1295: Odd parallelisation behaviour in frequencies by findif of gradient
During debugging of the failing (on Windows) test in #1235, I have noticed the following:
```
(psi4-dev-cbs_dict) kraus@dream:~/psi4/tests/cbs-xtpl-freq::psi4 -n 1 -i in.dat -o n1.out
	Nuclear repulsion energy..........................................PASSED
 6 displacements needed.
 1 2 3 4 5 6
	SCF/cc-pVDZ Frequency 1...........................................PASSED
	SCF/cc-pVDZ Frequency 2...........................................PASSED
	SCF/cc-pVDZ Frequency 3...........................................PASSED
(psi4-dev-cbs_dict) kraus@dream:~/psi4/tests/cbs-xtpl-freq::psi4 -n 4 -i in.dat -o n4.out
	Nuclear repulsion energy..........................................PASSED
 9 displacements needed.
 1 2 3 4 5 6 7 8 9
Warning: used thermodynamics relations inappropriate for low-frequency modes: ['476.8559' '479.2930' '484.2715']
	SCF/cc-pVDZ Frequency 1: computed value (1809.246) does not match (476.856) to 2 digits.
Traceback (most recent call last):
  File "/home/kraus/Applications/psi4-cbs_dict/bin/psi4", line 271, in <module>
    exec(content)
  File "<string>", line 34, in <module>
  File "/home/kraus/Applications/psi4-cbs_dict/lib/psi4/driver/p4util/util.py", line 227, in compare_values
    raise TestComparisonError(message)

TestComparisonError: 	SCF/cc-pVDZ Frequency 1: computed value (1809.246) does not match (476.856) to 2 digits.
```
The in.dat, n1.out and n2.out files are attached.
[in.txt](https://github.com/psi4/psi4/files/2488549/in.txt)
[n1.txt](https://github.com/psi4/psi4/files/2488550/n1.txt)
[n4.txt](https://github.com/psi4/psi4/files/2488551/n4.txt)

The number of SALC's  in the parallel version (see line 362) is wrong: it should be 3, it is 6. @JonathonMisiewicz , ideas?

### PR DESCRIPTION
**Pull request by PeterKraus**
## Description
Fixes #1295 .

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/libmints/matrix.cc [hunk 1]
```diff
@@ -1251,7 +1251,7 @@ double Matrix::rms() {
     double sum = (double)0.0;
     long terms = 0;
     for (int h = 0; h < nirrep_; ++h) {
-#pragma omp parallel for reduction(+ : sum)
+#pragma omp parallel for reduction(+ : sum, terms)
         for (int i = 0; i < rowspi_[h]; ++i) {
             for (int j = 0; j < colspi_[h ^ symmetry_]; ++j) {
                 sum += matrix_[h][i][j] * matrix_[h][i][j];
```
Comment (dgasmith) [inline psi4/src/psi4/libmints/matrix.cc]: Might be better to write the following above the OMP loop, but either way fixes the error.

```
terms += rowspi_[h] * colspi_[h ^ symmetry_];
```

Comment (PeterKraus) [reply psi4/src/psi4/libmints/matrix.cc]: Fair enough. I was also wondering about the loop below: the `absmax()` function can also be parallelised the same way (using custom resolution in OMP >=4.5). Which versions of OMP do we target?

Comment (dgasmith) [reply psi4/src/psi4/libmints/matrix.cc]: I don't think we target any version of OMP at the moment, when 5.0 comes out I think we will have to review that strategy. @loriab may have more comments.

Comment (loriab) [reply psi4/src/psi4/libmints/matrix.cc]: * Prior to the news that MSVC only supports OpenMP v2, I was scarcely aware of OpenMP versioning
* conda
  * linux: conda `gcc` `v7.3`, OpenMP is `v4.5`
  * linux: `icpc` `v2018.3` atop conda `gcc` `v7.3`, OpenMP is `v5.0`
  * mac: conda `clang` `v4.0.1`, OpenMP is `v3.1` whether `libomp` _or_ `libiomp5` (suggesting to me that the compiler is the constraint)
  * mac: `icpc` `v2018.2` atop conda clang `v4.0.1`, looks like I need to renew my license :-(

**Pull request merged**